clients/src/main/java/org/apache/kafka/clients/consumer/internals/AbstractCoordinator.java

```diff
@@ -852,14 +857,23 @@ public synchronized void maybeLeaveGroup(String leaveReason) {
             // attempt any resending if the request fails or times out.
             log.info("Member {} sending LeaveGroup request to coordinator {} due to {}",
                      generation.memberId, coordinator, leaveReason);
+
             LeaveGroupRequest.Builder request = new LeaveGroupRequest.Builder(new LeaveGroupRequestData()
-                    .setGroupId(rebalanceConfig.groupId).setMemberId(generation.memberId));
-            client.send(coordinator, request)
+                    .setGroupId(rebalanceConfig.groupId)
+                    // need to set both member.id and memberIdentity fields so that builder could choose
+                    // to build one of them in runtime.
+                    .setMemberId(generation.memberId)
```

Comment:
Could we push this logic into `LeaveGroupRequest.Builder`? Then in `build(version)`, we can just set the right field.